server/application/application.go

```diff
@@ -1897,6 +1905,22 @@ func (s *Server) GetApplicationSyncWindows(ctx context.Context, q *application.A
 	return res, nil
 }
 
+func (s *Server) inferResourcesStatusHealth(app *v1alpha1.Application) {
+	if !argocommon.PersistResourceStatusHealth {
```

Comment:
Should this boolean be part of the Application cache key?

The upside would be that, after someone toggles on the option, there's no moment when health statuses are _not_ returned.

The downside would be that, after someone toggles on the option, the entire Application cache is now out of date.

Reply:
I've got, hopefully, a good idea: introduced `resourceHealthSource` field in application status that indicates if resources health is in CRD (by default) or in application tree.

This eliminates the requirement to coordinate new flag value between application controller & API server. Another benefit is that health status will be gradually removed from CRDs by controller.